dnf5-plugins/config-manager_plugin/shared.hpp

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef DNF5_COMMANDS_CONFIG_MANAGER_CONFIG_MANAGER_SHARED_HPP
+#define DNF5_COMMANDS_CONFIG_MANAGER_CONFIG_MANAGER_SHARED_HPP
+
+#include <libdnf5/conf/config_main.hpp>
+#include <libdnf5/conf/const.hpp>
+#include <libdnf5/utils/bgettext/bgettext-mark-domain.h>
+
+#include <filesystem>
+
+namespace dnf5 {
+
+const std::filesystem::path CFG_MANAGER_REPOS_OVERRIDE_FILENAME = "99-config_manager.repo";
+
+const std::filesystem::path CFG_MANAGER_REPOS_OVERRIDE_FILEPATH =
+    libdnf5::REPOS_OVERRIDE_DIR / CFG_MANAGER_REPOS_OVERRIDE_FILENAME;
+
+struct ConfigManagerError : public libdnf5::Error {
+    using Error::Error;
+    const char * get_domain_name() const noexcept override { return "dnf5"; }
+    const char * get_name() const noexcept override { return "ConfigManagerError"; }
+};
+
+// Checks if the `path` directory exists. If not, according to the `create_missing_dirs` argument,
+// the directories (missing paths elements) are created or `ConfigManagerError` exception is thrown.
+inline void resolve_missing_dir(const std::filesystem::path & path, bool create_missing_dirs) {
+    if (!std::filesystem::exists(path)) {
```

Comment:
This is real nitpicking - if the path is a file, the test pass because the path exists.
```
$ mkdir /tmp/config-manager/etc/ -p
$ touch /tmp/config-manager/etc/dnf
$ dnf5 --installroot=/tmp/config-manager config-manager setopt gpgcheck=false
```
There is `std::filesystem::is_directory` method. But if the path is a `symlink` to an existing directory then it will not work according to expectation - pass.

In other cases it works perfectly.